server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -629,6 +653,48 @@ private void checkAliasNameConflicts(Map<String, String> renamedIndices, Set<Str
                         }
                     }
 
+                    private String[] getIgnoreSettingsInternal() {
+                        // for non-remote store enabled domain, we will remove all the remote store
+                        // related index settings present in the snapshot.
+                        String[] indexSettingsToBeIgnored = new String[] {};
+                        if (false == RemoteStoreNodeAttribute.isRemoteStoreAttributePresent(clusterService.getSettings())) {
+                            indexSettingsToBeIgnored = ArrayUtils.concat(indexSettingsToBeIgnored, new String[] { "index.remote_store.*" });
+                        }
+                        return indexSettingsToBeIgnored;
+                    }
+
+                    private Settings getOverrideSettingsInternal() {
+                        final Settings.Builder settingsBuilder = Settings.builder();
+                        updateReplicationStrategy(
+                            settingsBuilder,
+                            request.indexSettings(),
+                            clusterService.getSettings(),
+                            clusterService.getClusterSettings()
+                        );
+                        // remote store settings needs to be overridden if the remote store feature is enabled in the
+                        // cluster where snapshot is being restored.
+                        MetadataCreateIndexService.updateRemoteStoreSettings(settingsBuilder, clusterService.getSettings());
+                        return settingsBuilder.build();
+                    }
+
+                    private void updateReplicationStrategy(
```

Comment:
it looks like `MetadataCreateIndexService.updateReplicationStrategy` consider multiple things -
        // 1. Explicit index creation request parameter
        // 2. Template property for replication type
        // 3. custom Cluster replication type setting 
        // 4. Defaults to segment if remote store attributes on the cluster
        // 5. Default cluster level setting
Not  all of these applies in case of snapshot. However, @harishbhakuni  we could still refactor the original method to reuse it here?

Reply:
the difference between that one and this one is during restore we want to allow user to restore with the replication setting to whatever it was in snapshot's index metadata if customer do not have `cluster.index.restrict.replication.type` setting enabled. if this is enabled, we need to override the settings with the cluster default so that index can still be restored.